src/html/Quoting.cc

```diff
@@ -7,81 +7,51 @@
  */
 
 #include "squid.h"
+#include "base/CharacterSet.h"
 #include "html/Quoting.h"
 
-/*
- *  HTML defines these characters as special entities that should be quoted.
- */
-static struct {
-    unsigned char code;
-    const char *quote;
-} htmlstandardentities[] =
-
-{
-    /* NOTE: The quoted form MUST not be larger than 6 character.
-     * see close to the MemPool commend below
-     */
-    {
-        '<', "&lt;"
-    },
-    {
-        '>', "&gt;"
-    },
-    {
-        '"', "&quot;"
-    },
-    {
-        '&', "&amp;"
-    },
-    {
-        '\'', "&#39;"
-    },
-    {
-        0, NULL
-    }
-};
+#include <cstring>
+#include <unordered_map>
 
-/*
- *  html_do_quote - Returns a static buffer containing the quoted
- *  string.
- */
 char *
 html_quote(const char *string)
 {
```

Comment:
> https://www.w3.org/MarkUp/HTMLPlus/htmlplus_59.html#0 seems to be the closest I can find to an authority.

I think we should refer to https://html.spec.whatwg.org/#character-references or https://html.spec.whatwg.org/multipage/syntax.html#syntax-charref (which covers the general `&#n;` encoding mechanism and refers to https://html.spec.whatwg.org/multipage/named-characters.html#named-character-references for specific named character references).
 
> Quoting from https://www.w3.org/MarkUp/HTMLPlus/htmlplus_13.html: "There are a large number of entities defined by the ISO, covering most languages and symbols for publishing and mathematics. Requiring all browsers to support these would be impractical"

I do not think we care about that problem (in this PR scope) because we do not support UTF-8 or other standards that actually allow for that "larger number of entities". We only care about a handful of named character references we use and the general (decimal) numeric character reference mechanism for encoding bytes.

```suggestion
{
    // See https://html.spec.whatwg.org/#character-references
```

I do not insist on this change.

P.S. The references you found are for "HTML+ discussion documents". The references I recommend are for the HTML standard, albeit a "living" one.